Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp

```diff
@@ -799,6 +809,60 @@ WebKitWebViewAccessible* View::accessible() const
 }
 #endif
 
+#if ENABLE(WPE_PLATFORM)
+void View::handleGesture(WPEEvent* event)
+{
+    auto gestureController = wpe_view_get_gesture_controller(m_wpeView.get());
+    if (!gestureController)
+        return;
+
+    wpe_gesture_controller_handle_event(gestureController, event);
+
+    if (wpe_event_get_event_type(event) == WPE_EVENT_TOUCH_DOWN)
+        return;
+
+    switch (wpe_gesture_controller_get_gesture(gestureController)) {
+    case WPE_GESTURE_NONE:
+        break;
+    case WPE_GESTURE_TAP:
+        if (double x, y; wpe_gesture_controller_get_gesture_position(gestureController, &x, &y)) {
```

Comment:
I guess we only want to do this is the event is UP.

Reply:
Yes, there's check for that above.